version: '2.1'
services:
  app:
    image: fcollman/synapsedb
    ports:
      - 80:80
    volumes:
      - ${PWD}/docker/docker_compose_config.py:/synapsedb/synapsedb/instance/dev_config.py
      - ${PWD}/atbigdawg-nginx.conf:/etc/nginx/conf.d/flask-site-nginx.conf
      - /nas3:/nas3:ro
      - /nas5:/nas5:ro
      - /nas2:/nas2:ro
      - /nas:/nas:ro
      - /nas4:/nas4:ro
      - /nas6:/nas6:ro
      - ${PWD}:/synapsedb
    links:
      - db
    depends_on:
      db:
        condition: service_healthy
  db:
    image: mdillon/postgis
    environment:
      POSTGRES_PASSWORD: synapsedb
    ports:
      - 5432:5432
    volumes:
      - /mnt/SSD/postgresdata:/var/lib/postgresql/data
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -q -U postgres"
      interval: 3s
      timeout: 5s
      retries: 5
  adminer:
    image: adminer
    ports:
      - 8089:8080
    links:
      - db
